test/tools/yulopti.cpp

```diff
@@ -102,40 +115,51 @@ class YulOpti
 		size_t _columns
 	)
 	{
+		yulAssert(_columns > 0, "");
+
 		auto hasShorterString = [](auto const& a, auto const& b){ return a.second.size() < b.second.size(); };
-		size_t longestDescriptionLength = max(
+		size_t longestDescriptionLength = std::max(
 			max_element(_optimizationSteps.begin(), _optimizationSteps.end(), hasShorterString)->second.size(),
 			max_element(_extraOptions.begin(), _extraOptions.end(), hasShorterString)->second.size()
 		);
 
-		size_t index = 0;
-		auto printPair = [&](auto const& optionAndDescription)
-		{
-			cout << optionAndDescription.first << ": ";
-			cout << setw(static_cast<int>(longestDescriptionLength)) << setiosflags(ios::left);
-			cout << optionAndDescription.second << " ";
+		vector<string> overlappingAbbreviations =
+			ranges::views::set_intersection(_extraOptions | views::keys, _optimizationSteps | views::keys) |
+			views::transform([](char _abbreviation){ return string(1, _abbreviation); }) |
+			to<vector>();
 
-			++index;
-			if (index % _columns == 0)
-				cout << endl;
-		};
+		yulAssert(
+			overlappingAbbreviations.empty(),
+			"ERROR: Conflict between yulopti controls and the following Yul optimizer step abbreviations: " +
+			boost::join(overlappingAbbreviations, ", ") + ".\n"
+			"This is most likely caused by someone adding a new step abbreviation to "
+			"OptimiserSuite::stepNameToAbbreviationMap() and not realizing that it's used by yulopti.\n"
+			"Please update the code to use a different character and recompile yulopti."
+		);
 
-		for (auto const& optionAndDescription: _extraOptions)
+		auto optionCompare = [](tuple<char, string> const& _option1, tuple<char, string> const& _option2)
 		{
-			yulAssert(
-				_optimizationSteps.count(optionAndDescription.first) == 0,
-				"ERROR: Conflict between yulopti controls and Yul optimizer step abbreviations.\n"
-				"Character '" + string(1, optionAndDescription.first) + "' is assigned to both " +
-				optionAndDescription.second + " and " + _optimizationSteps.at(optionAndDescription.first) + " step.\n"
-				"This is most likely caused by someone adding a new step abbreviation to "
-				"OptimiserSuite::stepNameToAbbreviationMap() and not realizing that it's used by yulopti.\n"
-				"Please update the code to use a different character and recompile yulopti."
+			return (
+				_option1 != _option2 ?
+				boost::algorithm::lexicographical_compare(get<1>(_option1), get<1>(_option2), boost::algorithm::is_iless()) :
+				tolower(get<0>(_option1)) < tolower(get<0>(_option2))
```

Comment:
Isn't `option1 == option2` in this line?

Reply:
Ah, right! `_option1 != _option2` should have been a case-insensitive comparison. I'll fix that in a moment.

Reply:
Done.